olimex/teres_i/default.nix

```diff
@@ -0,0 +1,30 @@
+{ config, lib, pkgs, ...}:
+
+{
+  hardware.deviceTree = {
+    name = lib.mkDefault "allwinner/sun50i-a64-teres-i.dts";
+    enable = lib.mkDefault true;
+  };
+
+  boot = {
+    consoleLogLevel = lib.mkDefault 7;
+
+    kernelPackages = lib.mkDefault pkgs.linuxPackages_latest;
+    kernelParams = lib.mkDefault ["console=ttyS0,115200n8"];
+    extraModulePackages = lib.mkDefault [];
+
+    initrd = {
+      availableKernelModules = lib.mkDefault ["usbhid"];
+      kernelModules = lib.mkDefault [];
```

Comment:
This likely do nothing. See previous comment.

Reply:
> These likely do nothing. See previous comment. -- @samueldr (https://github.com/NixOS/nixos-hardware/pull/750#discussion_r1349257607)

Link "the previous comment" next time, it makes it hard to follow plz.

Assuming you meant this one: https://github.com/NixOS/nixos-hardware/pull/750#discussion_r1349559992

`usbhid` is what generates `nixos-generate-config` when called so i am trying to follow that.

the unsets for empty lists are an attempt to highlight to the end-user that these are relevant for hardware modifications (very common practice among teres users) and to keep them empty by default.

Reply:
> the unsets for empty lists are an attempt to highlight to the end-user that these are relevant for hardware modifications (very common practice among teres users) and to keep them empty by default.

See [this comment](https://github.com/NixOS/nixos-hardware/pull/750/files#r1349565088).

TLDR: it won't do anything as those are set with higher priority. Setting any list with `lib.mkDefault`, will not do what you expect here, and "setting" an "empty" list would not do what you expect here even if the priority was the same, with the merge semantics of lists.